src/main/kotlin/app/revanced/patches/youtube/layout/hide/getpremium/resource/patch/HideGetPremiumVideoAdvertisementResourcePatch.kt

```diff
@@ -0,0 +1,34 @@
+package app.revanced.patches.youtube.layout.hide.getpremium.resource.patch
+
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.ResourcePatch
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patches.shared.mapping.misc.patch.ResourceMappingPatch
+import app.revanced.patches.shared.settings.preference.impl.StringResource
+import app.revanced.patches.shared.settings.preference.impl.SwitchPreference
+import app.revanced.patches.youtube.layout.hide.getpremium.annotations.HideGetPremiumVideoAdvertisementCompatibility
+import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+
+@Name("hide-get-premium-resource-patch")
+@HideGetPremiumVideoAdvertisementCompatibility
+@DependsOn([SettingsPatch::class, ResourceMappingPatch::class])
+@Version("0.0.1")
+class HideGetPremiumVideoAdvertisementResourcePatch : ResourcePatch {
```

Comment:
Couple of things:

- Patches that don't face the user do not need a name or compatibility as the compatibility is transitive. 
- The dependency `ResourceMappingPatch` has slipped through.
- It is not really necessary to create a separate resource patch to use `SettingsPatch`. If the motivation was for consistency or semantical reasons, you can keep it.
- 
```suggestion
@Name("hide-get-premium-resource-patch")
@DependsOn([SettingsPatch::class])
class HideGetPremiumVideoAdvertisementResourcePatch : ResourcePatch {
```

Reply:
In the early days, this was done the way you did it for a short period of time, probably missed refactoring.